### ISSUE TITLE
Partials looks ahead :: Feature Request

### ISSUE THREAD

**Issue by xerocole** [CliffS/mustache-simple#7]
Hi @CliffS ,

Thanks for the great work on this, I was originally working with the amazing work done by the Template::Mustache module, but it was a bit more complex to work with than I had envisioned. I have just recently switched to your module and the performance it dramatically faster.

That said, I am working on a CMS style system that will try to leverage and Akamai "way of life" in terms of caching. In order to that do, I am looking at having some type of algorithm in place to generate a tag (or .cache.filename ) that will cache the output of a request. 

Pretty straight forward. The how is not so easily done.

In short, is there a simple way to get the actually file names of all the partials that a template (including nested) will contain? This would allow me to "look ahead" sort of speak at what a template requires (file-base wise) and get the last modified dates and potentially create a sum of all the epoch dates to see if any of the files have changed, making it necessary to generate a new .cache file.

This will need to be performant cause it will be handled at each server request.

Thoughts? and great work again, thanks for this.

**Comment by CliffS** [comment]
Let me have a think about this.  I can't help but think there's a better way than you're suggesting but it's not come to me yet,

**Comment by CliffS** [comment]
Actually, your idea won't work because, until the context is available, it's impossible to know whether a partial resolves to a file or to code.  Also the path taken may change based on the data.

What are you using as a server?  Assuming it's perl (speedy) via nginx on linux with plenty of RAM, the cost of opening and loading each partial is probably less than the cost of stat'ing it and then cacheing.

**Comment by xerocole** [comment]
@CliffS Thanks, that was what I was reading as well in the code base, but was a bit hopeful you had a trick up your sleeve 👍 . 

That said, the server is using Fast::CGI so pretty speedy already, and originally I am just making a blanket cache system which stales after a few minutes (5 mins) so every request gets a cache and after 5 minutes it get refreshed. This does the trick for the moment but was hoping to get something more intelligent so just coming up with new approaches to a bit of an old problem.

Initially, I started with Template::Mustache since I was sub-classing a more hybrid parser, but then things got a bit complicated in terms of abstraction too quickly. So I think I may just incorporate a "partials" only parser that will generate the code before the rest is handed off the Mustache::Simple to render.

If there are not issues, as the author, I can also try to add a `->lookahead()` function to the class. And can PR it if you are willing. Or if you prefer I can add it as a `->etag()` function that does what I was looking for (*that was my original code I was doing to Template::Mustache, trying to create a cache compatible tag)

Either way, a super big thanks thou and love the module, keep it up, it is really performant and well written.

**Comment by CliffS** [comment]
Very happy with any PR that doesn't break he codebase. :)

**Comment by akrabat** [comment]
The author of this component died in 2022 and this repository is now archived.
